### PR TITLE
Stats: Fix `stats_insights_accessed` initial section tracking

### PR DESCRIPTION
**Pull request by ashiagr**
This PR fixes `stats_insights_accessed` initial section tracking when `initialSection` is not passed to `StatsViewModel`.

**Details**

When `initialSection` is not passed to `StatsViewModel`, the default tab selection is not tracked until now (see p1648141247441739/1648128476.399289-slack-C011BKNU1V5). This issue is present for an unknown time (possibly since the beginning of `Stats` implementation). 

Since `INSIGHTS` is the first tab in the `Stats` screen's list of tabs, if `initialSection` is not passed to `StatsViewModel`, we now track the default section as `INSIGHTS` section.

**To test**

Test.1  From `Stats Quick Link`/`Stats Menu` 

1. Open the app. 
2. Enable logging from `Me` -> `App Settings` -> `Privacy Settings` -> `Collect information ON`.
3. Click `Stats Quick Link` or `Stats Menu` from the `My Site` tab
4. Notice that Stats -> `INSIGHTS` section is shown.
5. Notice in logs: 
   - 🔵 Tracked: stats_accessed (with properties)
   - 🔵 Tracked: stats_insights_accessed
6. Toggle to other sections. Return to `INSIGHTS` section.
7.  Notice that logs have below tracking at the end: 
    - 🔵 Tracked: stats_insights_accessed

Test.2 From `Stats Shortcut`

1. Open stats from app's shortcut (right click app icon and click Stats shorcut)
2. Notice that Stats -> `INSIGHTS` section is shown.
3. Notice in logs: 
   - 🔵 Tracked: stats_accessed (with properties)
   - 🔵 Tracked: stats_insights_accessed
  
Test.3 From `Stats Widgets`

1. Add `Stats Views Widget` to the device and select a site to view.
2. Click the widget.
3. Notice that Stats -> `INSIGHTS` section is shown.
5. Notice in logs: 
   - 🔵 Tracked: stats_accessed (with properties)
   - 🔵 Tracked: stats_insights_accessed
   - 🔵 Tracked: stats_widget_tapped (with properties)

**Merging Notes**

Since the initial `INSIGHTS` section is not tracked for an unknown time, and not something that impacted `stats_insights_accessed` event trend since `release/19.3`, let's wait for approval to merge it now or later.

- Make sure that https://github.com/wordpress-mobile/WordPress-Android/pull/16169 is merged.
- Wait for approval to merge to `release/19.5`.
- Remove `Not Ready for Merge` label.
- Merge the PR.

Targets `release/19.5`
/cc @AliSoftware 

## Regression Notes
1. Potential unintended areas of impact
Stats event tracking from Widgets and Shorcuts.

4. What I did to test those areas of impact (or what existing automated tests I relied on)
See **To Test** section.

6. What automated tests I added (or what prevented me from doing so) N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ashiagr** [comment]
@leandroalonso let us know if it is the right time to merge this PR. I'm asking because `stats_insights_accessed` event trend will get a considerable bump as the issue existed for a long a time (possibly since the beginning), so any monitoring of this event at this time might be impacted.

**Review by ravishanker — APPROVED** [review]
LGTM 👍  Thanks for fixing it quickly 🚀

**Comment by AliSoftware** [comment]
@ashiagr Thanks for the fix!

> Merging Notes
>
> Since the initial INSIGHTS section is not tracked for an unknown time, and not something that impacted stats_insights_accessed event trend since release/19.3, let's wait for approval to merge it now or later.
>
> - Make sure that https://github.com/wordpress-mobile/WordPress-Android/pull/16169 is merged.
> - Wait for approval to merge to release/19.5.
> - Remove Not Ready for Merge label.
> - Merge the PR.

This should now be ready to merge this, since I just merged #16169, and @ravishanker already approved this PR.

I'm still going to wait for @leandroalonso 's response [on this question about the timing](https://github.com/wordpress-mobile/WordPress-Android/pull/16171#issuecomment-1078725231) first before we get the go to merge it and I issue a new beta later today.

**Comment by leandroalonso** [comment]
> let us know if it is the right time to merge this PR. I'm asking because stats_insights_accessed event trend will get a considerable bump as the issue existed for a long a time (possibly since the beginning), so any monitoring of this event at this time might be impacted.

@ashiagr @AliSoftware I think this was a bug, so we should be good to merge. Without it, we would be able to check the difference but that wouldn't reflect the real data.

(@yaelirub FYI: we're going to merge a bug fix regarding `stats_insights_accessed`. This event wasn't fired correctly before, thus this was undertracked. With this fix we'll have a bump on it — which is expected).

**Comment by AliSoftware** [comment]
This fix has now been shipped with `19.5-rc-2` 🎉

**Pull request merged**
